Sources_v3/Database/DTOs/ChannelListQueryDTO.swift

```diff
@@ -12,7 +12,7 @@ class ChannelListQueryDTO: NSManagedObject {
     @NSManaged var filterHash: String
     
     /// Serialized `Filter` JSON which can be used in cases the query needs to be repeated, i.e. for newly created channels.
-    @NSManaged var filterJSONData: Data
+    @NSManaged var filterJSONData: Data?
```

Comment:
This can't be optional. We must be able to always store the serialized filter, no?

Reply:
It should have filterJSONData, right.
I just don't want this `Data` to be overwritten. 
As far as I understand we can save ChannelListQueryDTO only through function that is inside `extension NSManagedObjectContext` and we are trying to keep this rule regarding all DTOs. In this case I believe it's safe enough.

Reply:
I see. In that case, I'd suggest doing just:
```swift
    func saveQuery(query: ChannelListQuery) -> ChannelListQueryDTO {
        if let existingDTO = ChannelListQueryDTO.load(filterHash: query.filter.filterHash, context: self) {
            return existingDTO
        }
        ...
```
to prevent orerriding the query, if the entity already exists.